src/apps/ocioview/ocioview/inspect/cube_inspector.py

```diff
@@ -0,0 +1,789 @@
+
+    @QtCore.Slot(bool)
+    def _on_show_src_cube_button_clicked(self, checked: bool) -> None:
+        """Toggle whether the source cube edges are visible."""
+        self.view.set_show_source_cube(checked)
+
+    @QtCore.Slot(bool)
+    def _on_clamp_to_src_cube_button_clicked(self, checked: bool) -> None:
+        """Toggle whether to clamp samples to source cube."""
+        self.view.set_clamp_to_source_cube(checked)
+
+    @QtCore.Slot(float, float, float)
+    def _on_sample_changed(
+        self, r_output: float, g_output: float, b_output: float
+    ) -> None:
+        """Update color value labels from currently hovered sample."""
+        self.output_r_sample_label.setText(
+            self.FMT_R_LABEL.format(v=self._sample_format.format(v=r_output))
+        )
+        self.output_g_sample_label.setText(
+            self.FMT_G_LABEL.format(v=self._sample_format.format(v=g_output))
+        )
+        self.output_b_sample_label.setText(
+            self.FMT_B_LABEL.format(v=self._sample_format.format(v=b_output))
+        )
+        self.output_sample_swatch.setStyleSheet(
+            self.FMT_SWATCH_CSS.format(
+                r=float_to_uint8(r_output),
+                g=float_to_uint8(g_output),
+                b=float_to_uint8(b_output),
+            )
+        )
+
+    @QtCore.Slot(int)
+    def _on_sample_precision_changed(self, value: float) -> None:
+        """Update sample label decimal precision."""
+        self._sample_format = f"{{v:.{value}f}}"
+
+
+class CubeView(QtWidgets.QWidget):
+    """Widget for rendering a color volume point cloud."""
+
+    sample_changed = QtCore.Signal(float, float, float)
+
+    CUBE_SIZE_DEFAULT = 33
+    IMAGE_DETAIL_DEFAULT = 4
+
+    def __init__(
+        self,
+        input_src: InputSource = InputSource.CUBE,
+        cube_size: int = CUBE_SIZE_DEFAULT,
+        image_detail: int = IMAGE_DETAIL_DEFAULT,
+        show_src_cube: bool = True,
+        clamp_to_src: bool = False,
+        parent: Optional[QtWidgets.QWidget] = None,
+    ):
+        """
+        :param input_src: Source for plotted samples
+        :param cube_size: Number of samples per cube dimension
+        :param image_detail: Ratio of image samples to pixels
+        :param show_src_cube: Whether to show source cube edges
+        :param clamp_to_src: Whether to clamp samples to source cube
+        """
+        super().__init__(parent=parent)
+        self.setMouseTracking(True)
+        self.setSizePolicy(
+            QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Expanding
+        )
+        size = self.size()
+
+        self._input_src = input_src
+        self._cube_size = cube_size
+        self._image_detail = image_detail
+        self._show_src_cube = show_src_cube
+        self._clamp_to_src = clamp_to_src
+
+        self._mouse_button = None
+        self._mouse_last_pos = None
+        self._cpu_proc = None
+        self._image_buf: np.ndarray = None
```

Comment:
Type is `oiio.ImageBuffer` here I think.

Reply:
Good catch. Thanks!